In [80]:
import os
import sys
import pprint
import pandas as pd

base_dir = 'F:\Thesis_ssd\MasterThesis3.0'
os.chdir(base_dir)

from Classes.DataProcessing.LoadData import LoadData
from Classes.DataProcessing.DataHandler import DataHandler
from Classes.DataProcessing.DataGenerator import DataGenerator
from Classes.Modeling.GridSearchResultProcessor import GridSearchResultProcessor
from Classes.Modeling.CustomCallback import CustomCallback
from Classes.Scaling.ScalerFitter import ScalerFitter
from Classes.Scaling.MinMaxScalerFitter import MinMaxScalerFitter
from Classes.Scaling.StandardScalerFitter import StandardScalerFitter
from Classes.Modeling.ResultFitter import ResultFitter
import json

In [16]:
load_args = {
    'earth_explo_only' : True,
    'noise_earth_only' : False,
    'downsample' : True,
    'upsample' : True,
    'frac_diff' : 0.5,
    'seed' : 1,
    'subsample_size' : 0.25
}

loadData = LoadData(**load_args)

full_ds, train_ds, val_ds, test_ds = loadData.get_datasets()
noise_ds = loadData.noise_ds
handler = DataHandler(loadData)
dataGen = DataGenerator(loadData)
resultFitter = ResultFitter(loadData)

AttributeError: 'LoadData' object has no attribute 'getDatasets'

In [6]:
result_file_name = 'results_6_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
model = resultFitter.fit_from_csv_and_index(result_file_name, 0, 2, 10, use_tensorboard = False, 
                               use_liveplots = True, use_custom_callback = False)

NameError: name 'resultFitter' is not defined

In [244]:
def f1_score(precision, recall):
    f1 = 2*((precision*recall)/(precision + recall))
    return f1

def create_f1_list(precision_df, recall_df):
    f1 = []
    for i in range(len(precision_df)):
        f1.append(f1_score(precision_df.loc[i], recall_df.loc[i]))
    return f1

result_file_name = 'results_LSTM_NARROW_noiseNotNoise_timeAug_sscale_noiseAug_earlyS.csv'
df = GridSearchResultProcessor().get_results_df_by_name(result_file_name, 2)
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)

df_f1 = df.copy()
df_f1.columns=df_f1.columns.str.strip()
all_train_precision = df_f1['train_precision']
all_train_recall = df_f1['train_recall']
all_val_precision = df_f1['val_precision']
all_val_recall = df_f1['val_recall']
f1_train = create_f1_list(all_train_precision, all_train_recall)
f1_val = create_f1_list(all_val_precision, all_val_recall)
df_f1['train_f1'] = f1_train
df_f1['val_f1'] = f1_val
df_f1

batch_size  epochs  learning_rate  num_channels  num_layers optimizer  \
0           32      40          0.010             3           2       sgd   
1           32      40          0.010             3           2       sgd   
2           32      40          0.010             3           2       sgd   
3           32      40          0.100             3           2       sgd   
4           32      40          0.050             3           2       sgd   
5           32      40          0.025             3           2       sgd   
6           32      40          0.005             3           2       sgd   
7           24      40          0.010             3           2       sgd   
8           28      40          0.010             3           2       sgd   
9           34      40          0.010             3           2       sgd   
10          36      40          0.010             3           2       sgd   
11          32      40          0.010             3           2       sgd   
12          32      40          0.010             3           2       sgd   
13          32      40          0.010             3           2       sgd   
14          32      40          0.010             3           2       sgd   
15          32      40          0.010             3           2       sgd   
16          32      40          0.010             3           2       sgd   
17          32      40          0.010             3           2       sgd   
18          32      40          0.010             3           2       sgd   
19          32      40          0.010             3           2       sgd   
20          32      40          0.010             3           2       sgd   
21          32      40          0.010             3           2       sgd   
22          32      40          0.010             3           2       sgd   
23          32      40          0.010             3           2       sgd   
24          32      40          0.010             3           2       sgd   
25          32      40          0.010             3           2       sgd   
26          32      40          0.010             3           3       sgd   
27          32      40          0.010             3           4       sgd   
28          32      40          0.010             3           5       sgd   
29          32      40          0.010             3           6       sgd   
30          32      40          0.010             3           2       sgd   
31          32      40          0.010             3           2       sgd   
32          32      40          0.010             3           2       sgd   
33          32      40          0.010             3           2       sgd   

   activation       decay_sequence  dropout_rate  filters  kernel_size  \
0        tanh               [1, 2]          0.30       17            5   
1        tanh               [1, 2]          0.30       17            5   
2        tanh               [1, 2]          0.30       17            5   
3        tanh               [1, 2]          0.30       17            5   
4        tanh               [1, 2]          0.30       17            5   
5        tanh               [1, 2]          0.30       17            5   
6        tanh               [1, 2]          0.30       17            5   
7        tanh               [1, 2]          0.30       17            5   
8        tanh               [1, 2]          0.30       17            5   
9        tanh               [1, 2]          0.30       17            5   
10       tanh               [1, 2]          0.30       17            5   
11       tanh               [1, 2]          0.30       17            5   
12       tanh               [1, 4]          0.30       17            5   
13       tanh             [1, 0.5]          0.30       17            5   
14       tanh               [1, 1]          0.30       17            5   
15       tanh               [1, 2]          0.30       17            5   
16       tanh               [1, 2]          0.30  

In [245]:
sorted_f1_df = df_f1.sort_values(by=['val_f1', 'train_f1'], axis = 0, ascending=False)
sorted_f1_df

batch_size  epochs  learning_rate  num_channels  num_layers optimizer  \
4           32      40          0.050             3           2       sgd   
2           32      40          0.010             3           2       sgd   
16          32      40          0.010             3           2       sgd   
31          32      40          0.010             3           2       sgd   
20          32      40          0.010             3           2       sgd   
10          36      40          0.010             3           2       sgd   
22          32      40          0.010             3           2       sgd   
3           32      40          0.100             3           2       sgd   
32          32      40          0.010             3           2       sgd   
25          32      40          0.010             3           2       sgd   
24          32      40          0.010             3           2       sgd   
23          32      40          0.010             3           2       sgd   
11          32      40          0.010             3           2       sgd   
15          32      40          0.010             3           2       sgd   
18          32      40          0.010             3           2       sgd   
21          32      40          0.010             3           2       sgd   
1           32      40          0.010             3           2       sgd   
7           24      40          0.010             3           2       sgd   
14          32      40          0.010             3           2       sgd   
26          32      40          0.010             3           3       sgd   
19          32      40          0.010             3           2       sgd   
13          32      40          0.010             3           2       sgd   
12          32      40          0.010             3           2       sgd   
17          32      40          0.010             3           2       sgd   
8           28      40          0.010             3           2       sgd   
27          32      40          0.010             3           4       sgd   
0           32      40          0.010             3           2       sgd   
9           34      40          0.010             3           2       sgd   
5           32      40          0.025             3           2       sgd   
30          32      40          0.010             3           2       sgd   
6           32      40          0.005             3           2       sgd   
28          32      40          0.010             3           5       sgd   
29          32      40          0.010             3           6       sgd   
33          32      40          0.010             3           2       sgd   

   activation       decay_sequence  dropout_rate  filters  kernel_size  \
4        tanh               [1, 2]          0.30       17            5   
2        tanh               [1, 2]          0.30       17            5   
16       tanh               [1, 2]          0.30       17            5   
31       tanh               [1, 2]          0.25       17            5   
20       tanh               [1, 2]          0.30       17            5   
10       tanh               [1, 2]          0.30       17            5   
22       tanh               [1, 2]          0.30       17            5   
3        tanh               [1, 2]          0.30       17            5   
32       tanh               [1, 2]          0.20       17            5   
25       tanh               [1, 2]          0.30       17            5   
24       tanh               [1, 2]          0.30       17            5   
23       tanh               [1, 2]          0.30       17            5   
11       tanh               [1, 2]          0.30       17            5   
15       tanh               [1, 2]          0.30       17            5   
18       tanh               [1, 2]          0.30       17            5   
21       tanh               [1, 2]          0.30       17            5   
1        tanh               [1, 2]          0.30  

In [240]:
sorted_f1_df = df_f1.sort_values(by=['val_precision', 'train_precision'], axis = 0, ascending=False)
sorted_f1_df

batch_size  epochs  learning_rate  num_channels  num_layers optimizer  \
20          32      40          0.010             3           2       sgd   
2           32      40          0.010             3           2       sgd   
16          32      40          0.010             3           2       sgd   
24          32      40          0.010             3           2       sgd   
11          32      40          0.010             3           2       sgd   
10          36      40          0.010             3           2       sgd   
25          32      40          0.010             3           2       sgd   
23          32      40          0.010             3           2       sgd   
31          32      40          0.010             3           2       sgd   
18          32      40          0.010             3           2       sgd   
21          32      40          0.010             3           2       sgd   
22          32      40          0.010             3           2       sgd   
7           24      40          0.010             3           2       sgd   
19          32      40          0.010             3           2       sgd   
12          32      40          0.010             3           2       sgd   
9           34      40          0.010             3           2       sgd   
26          32      40          0.010             3           3       sgd   
14          32      40          0.010             3           2       sgd   
1           32      40          0.010             3           2       sgd   
27          32      40          0.010             3           4       sgd   
8           28      40          0.010             3           2       sgd   
6           32      40          0.005             3           2       sgd   
5           32      40          0.025             3           2       sgd   
13          32      40          0.010             3           2       sgd   
17          32      40          0.010             3           2       sgd   
30          32      40          0.010             3           2       sgd   
3           32      40          0.100             3           2       sgd   
4           32      40          0.050             3           2       sgd   
28          32      40          0.010             3           5       sgd   
29          32      40          0.010             3           6       sgd   
0           32      40          0.010             3           2       sgd   
15          32      40          0.010             3           2       sgd   
32          32      40          0.010             3           2       sgd   

   activation       decay_sequence  dropout_rate  filters  kernel_size  \
20       tanh               [1, 2]          0.30       17            5   
2        tanh               [1, 2]          0.30       17            5   
16       tanh               [1, 2]          0.30       17            5   
24       tanh               [1, 2]          0.30       17            5   
11       tanh               [1, 2]          0.30       17            5   
10       tanh               [1, 2]          0.30       17            5   
25       tanh               [1, 2]          0.30       17            5   
23       tanh               [1, 2]          0.30       17            5   
31       tanh               [1, 2]          0.25       17            5   
18       tanh               [1, 2]          0.30       17            5   
21       tanh               [1, 2]          0.30       17            5   
22       tanh               [1, 2]          0.30       17            5   
7        tanh               [1, 2]          0.30       17            5   
19       tanh               [1, 2]          0.30       17            5   
12       tanh               [1, 4]          0.30       17            5   
9        tanh               [1, 2]          0.30       17            5   
26       tanh            [1, 2, 4]          0.30       17            5   
14       tanh               [1, 1]          0.30     

In [236]:
df[df.columns[:15]].iloc[40:]

Empty DataFrame
Columns: [batch_size, epochs, learning_rate, num_channels, num_layers, optimizer, activation, decay_sequence, dropout_rate, filters, kernel_size, l1_r, l2_r, output_layer_activation, padding]
Index: []

In [160]:
df[df.columns[15:]].iloc[40:]

train_loss  train_precision  train_accuracy  train_recall  val_loss  \
40  1.084770    0.782372         0.786742        0.937814      1.147155   
41  0.899954    0.950041         0.860890        0.833142      1.037570   
42  1.925948    0.747005         0.749432        0.938575      1.981802   
43  2.269218    0.669505         0.519271        0.535989      2.274575   
44  2.271817    0.659082         0.502983        0.511301      2.273199   
45  0.806053    0.964770         0.904025        0.887121      1.025919   
46  0.841913    0.955342         0.908333        0.903420      1.085873   
47  0.910195    0.951840         0.911742        0.912532      1.132424   
48  0.988093    0.953485         0.908523        0.905616      1.212564   
49  1.054804    0.920381         0.908570        0.943146      1.275719   
50  0.840761    0.950863         0.906203        0.904704      1.066209   
51  0.862765    0.942918         0.909186        0.918035      1.087199   
52  0.926168    0.922502         0.896070        0.919923      1.150106   
53  0.884219    0.938466         0.909233        0.923055      1.129866   
54  0.650258    0.882517         0.948248        0.869578      0.858929   
55  0.654292    0.872542         0.944278        0.857623      0.836891   
56  0.737386    0.856420         0.951496        0.824561      0.851881   
57  0.769972    0.820915         0.943091        0.775694      0.875669   
58  0.779759    0.867669         0.970735        0.824440      0.958926   
59  0.669458    0.909886         0.950785        0.910667      0.889692   
60  0.683251    0.909164         0.976266        0.883891      0.920121   
61  0.675171    0.909095         0.947024        0.913430      0.918796   
62  1.118361    0.801154         0.809020        0.914703      1.173814   
63  0.941541    0.837935         0.953759        0.793012      1.041114   
64  0.869639    0.864079         0.958802        0.829813      1.041215   
65 NaN         NaN              NaN             NaN           NaN         

    val_precision  val_accuracy  val_recall  
40  0.755786       0.754972      0.923161    
41  0.891613       0.777936      0.751359    
42  0.707387       0.688447      0.891990    
43  0.654939       0.509470      0.526278    
44  0.650253       0.501420      0.512142    
45  0.893439       0.792614      0.774918    
46  0.875353       0.787405      0.786517    
47  0.878223       0.798295      0.802465    
48  0.877102       0.792850      0.794128    
49  0.836668       0.785038      0.833635    
50  0.869029       0.785748      0.791229    
51  0.861698       0.792614      0.812976    
52  0.840525       0.776515      0.811888    
53  0.864169       0.788589      0.802465    
54  0.781960       0.873274      0.779268    
55  0.785511       0.890763      0.765495    
56  0.771070       0.888552      0.742660    
57  0.736032       0.879151      0.690830    
58  0.777462       0.909132      0.732512    
59  0.800615       0.874074      0.811731    
60  0.802894       0.908358      0.776851    
61  0.788462       0.863495      0.803630    
62  0.762547       0.779974      0.886553    
63  0.783617       0.912383      0.739761    
64  0.777699       0.904444      0.737586    
65 NaN            NaN           NaN

## Findings:
 - New optimal dropout_rate: 0.2 (train_acc: 98%, val_acc: 90%) Old 0.3 (train_acc: 96.8%, val_acc: 90.9%)
 - RELU and Sigmoid activation functions do not work with this model.
 - L1 regularization is counter productive (Less is more) so let l1 = 0.00
 - 32 start neurons appear to be optimal

In [161]:
df.iloc[40:]

batch_size  epochs  learning_rate  num_layers optimizer activation  \
40  64          33      0.20000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
43  64          33      0.00010        2           sgd       tanh        
44  64          33      0.00001        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
53  64          33      0.01000        2           sgd       tanh        
54  32          33      0.01000        3           sgd       tanh        
55  32          33      0.01000        4           sgd       tanh        
56  32          33      0.01000        5           sgd       tanh        
57  32          33      0.01000        6           sgd       tanh        
58  24          33      0.01000        2           sgd       tanh        
59  28          33      0.01000        2           sgd       tanh        
60  34          33      0.01000        2           sgd       tanh        
61  36          33      0.01000        2           sgd       tanh        
62  32          33      0.10000        2           sgd       tanh        
63  32          33      0.05000        2           sgd       tanh        
64  32          33      0.02500        2           sgd       tanh        
65  32          33      0.00500        2           sgd       tanh        

         decay_sequence  dropout_rate  filters  kernel_size    l1_r   l2_r  \
40  [1, 2]               0.3           17       5            0.0001  0.001   
41  [1, 2]               0.3           17       5            0.0001  0.001   
42  [1, 2]               0.3           17       5            0.0001  0.001   
43  [1, 2]               0.3           17       5            0.0001  0.001   
44  [1, 2]               0.3           17       5            0.0001  0.001   
45  [1, 2]               0.3           17       5            0.0001  0.001   
46  [1, 2]               0.3           17       5            0.0001  0.001   
47  [1, 2]               0.3           17       5            0.0001  0.001   
48  [1, 2]               0.3           17       5            0.0001  0.001   
49  [1, 2]               0.3           17       5            0.0001  0.001   
50  [1, 2]               0.3           17       5            0.0001  0.001   
51  [1, 4]               0.3           17       5            0.0001  0.001   
52  [1, 0.5]             0.3           17       5            0.0001  0.001   
53  [1, 1]               0.3           17       5            0.0001  0.001   
54  [1, 2, 4]            0.3           17       5            0.0001  0.001   
55  [1, 2, 4, 6]         0.3           17       5            0.0001  0.001   
56  [1, 2, 4, 6, 8]      0.3           17       5            0.0001  0.001   
57  [1, 2, 4, 6, 8, 10]  0.3           17       5            0.0001  0.001   
58  [1, 2]               0.3           17       5            0.0001  0.001   
59  [1, 2]               0.3           17       5            0.0001  0.001   
60  [1, 2]               0.3           17       5            0.0001  0.001   
61  [1, 2]               0.3           17       5            0.0001  0.001   
62  [1, 2]               0.3           17       5            0.0001  0.001   
63  [1, 2]               0.3           17       5            0.0001  0.001   
64  [1, 2]               0.3           17       5      

In [162]:
def f1_score(precision, recall):
    f1 = 2*((precision*recall)/(precision + recall))
    return f1

def create_f1_list(precision_df, recall_df):
    f1 = []
    for i in range(len(precision_df)):
        f1.append(f1_score(precision_df.loc[i], recall_df.loc[i]))
    return f1

batch_size  epochs  learning_rate  num_layers optimizer activation  \
0   64          33      0.01000        2           sgd       tanh        
1   64          33      0.01000        2           sgd       tanh        
2   64          33      0.01000        2           sgd       tanh        
3   64          33      0.01000        2           sgd       tanh        
4   64          33      0.01000        2           sgd       tanh        
5   64          33      0.01000        2           sgd       tanh        
6   64          33      0.01000        2           sgd       tanh        
7   64          33      0.01000        2           sgd       tanh        
8   64          33      0.01000        2           adam      tanh        
9   64          33      0.01000        2           rmsprop   tanh        
10  64          33      0.01000        2           sgd       tanh        
11  64          33      0.01000        2           sgd       tanh        
12  64          33      0.01000        2           sgd       tanh        
13  64          33      0.01000        2           sgd       tanh        
14  64          33      0.01000        2           sgd       tanh        
15  64          33      0.01000        2           sgd       tanh        
16  64          33      0.01000        2           sgd       tanh        
17  64          33      0.01000        2           sgd       tanh        
18  64          29      0.01000        2           sgd       tanh        
19  64          37      0.01000        2           sgd       tanh        
20  64          33      0.01000        2           sgd       tanh        
21  64          33      0.01000        2           sgd       tanh        
22  64          33      0.01000        2           sgd       tanh        
23  64          33      0.01000        2           sgd       tanh        
24  64          33      0.01000        2           sgd       tanh        
25  64          33      0.01000        2           sgd       tanh        
26  64          33      0.01000        2           sgd       relu        
27  64          33      0.01000        2           sgd       sigmoid     
28  64          33      0.01000        2           sgd       softmax     
29  64          33      0.01000        2           sgd       tanh        
30  64          33      0.01000        2           sgd       tanh        
31  64          33      0.01000        1           sgd       tanh        
32  64          33      0.01000        3           sgd       tanh        
33  64          33      0.01000        4           sgd       tanh        
34  64          33      0.01000        5           sgd       tanh        
35  64          33      0.01000        6           sgd       tanh        
36  32          33      0.01000        2           sgd       tanh        
37  128         33      0.01000        2           sgd       tanh        
38  256         33      0.01000        2           sgd       tanh        
39  512         33      0.01000        2           sgd       tanh        
40  64          33      0.20000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
43  64          33      0.00010        2           sgd       tanh        
44  64          33      0.00001        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
53  64  

In [170]:
sorted_f1_df

batch_size  epochs  learning_rate  num_layers optimizer activation  \
19  64          37      0.01000        2           sgd       tanh        
10  64          33      0.01000        2           sgd       tanh        
25  64          33      0.01000        2           sgd       tanh        
11  64          33      0.01000        2           sgd       tanh        
13  64          33      0.01000        2           sgd       tanh        
47  64          33      0.01000        2           sgd       tanh        
36  32          33      0.01000        2           sgd       tanh        
51  64          33      0.01000        2           sgd       tanh        
14  64          33      0.01000        2           sgd       tanh        
29  64          33      0.01000        2           sgd       tanh        
49  64          33      0.01000        2           sgd       tanh        
48  64          33      0.01000        2           sgd       tanh        
37  128         33      0.01000        2           sgd       tanh        
3   64          33      0.01000        2           sgd       tanh        
53  64          33      0.01000        2           sgd       tanh        
12  64          33      0.01000        2           sgd       tanh        
32  64          33      0.01000        3           sgd       tanh        
40  64          33      0.20000        2           sgd       tanh        
4   64          33      0.01000        2           sgd       tanh        
45  64          33      0.01000        2           sgd       tanh        
18  64          29      0.01000        2           sgd       tanh        
46  64          33      0.01000        2           sgd       tanh        
50  64          33      0.01000        2           sgd       tanh        
5   64          33      0.01000        2           sgd       tanh        
52  64          33      0.01000        2           sgd       tanh        
2   64          33      0.01000        2           sgd       tanh        
34  64          33      0.01000        5           sgd       tanh        
0   64          33      0.01000        2           sgd       tanh        
6   64          33      0.01000        2           sgd       tanh        
62  32          33      0.10000        2           sgd       tanh        
7   64          33      0.01000        2           sgd       tanh        
38  256         33      0.01000        2           sgd       tanh        
41  64          33      0.10000        2           sgd       tanh        
15  64          33      0.01000        2           sgd       tanh        
35  64          33      0.01000        6           sgd       tanh        
59  28          33      0.01000        2           sgd       tanh        
24  64          33      0.01000        2           sgd       tanh        
1   64          33      0.01000        2           sgd       tanh        
16  64          33      0.01000        2           sgd       tanh        
27  64          33      0.01000        2           sgd       sigmoid     
23  64          33      0.01000        2           sgd       tanh        
28  64          33      0.01000        2           sgd       softmax     
22  64          33      0.01000        2           sgd       tanh        
21  64          33      0.01000        2           sgd       tanh        
20  64          33      0.01000        2           sgd       tanh        
26  64          33      0.01000        2           sgd       relu        
61  36          33      0.01000        2           sgd       tanh        
17  64          33      0.01000        2           sgd       tanh        
60  34          33      0.01000        2           sgd       tanh        
39  512         33      0.01000        2           sgd       tanh        
42  64          33      0.00100        2           sgd       tanh        
9   64          33      0.01000        2           rmsprop   tanh        
33  64          33      0.01000        4           sgd       tanh        
54  32  

In [155]:
pd.to_numeric(df_f1['val_f1'], errors='coerce')

0     0.821132
1     0.804132
2     0.824117
3     0.832628
4     0.830515
5     0.828167
6     0.820769
7     0.818637
8     0.766957
9     0.785517
10    0.850172
11    0.843275
12    0.831845
13    0.840825
14    0.836419
15    0.809115
16    0.800293
17    0.790123
18    0.828972
19    0.856512
20    0.796238
21    0.796238
22    0.796238
23    0.796238
24    0.805288
25    0.845431
26    0.796238
27    0.796238
28    0.796238
29    0.836253
30   NaN      
31    0.761004
32    0.831517
33    0.784201
34    0.821750
35    0.806797
36    0.837878
37    0.833272
38    0.818396
39    0.789155
40    0.831131
41    0.815500
42    0.789035
43    0.583601
44    0.572993
45    0.829969
46    0.828561
47    0.838636
48    0.833555
49    0.835149
50    0.828306
51    0.836628
52    0.825959
53    0.832174
54    0.780612
55    0.775374
56    0.756599
57    0.712715
58    0.754318
59    0.806134
60    0.789657
61    0.795973
62    0.819888
63    0.761058
64   NaN      
Name: val_f1, dtype: floa

In [152]:
df_f1['val_f1'].isnull().values.any()

True

In [156]:
df_f1.columns

Index(['batch_size', 'epochs', 'learning_rate', 'num_layers', 'optimizer',
       'activation', 'decay_sequence', 'dropout_rate', 'filters',
       'kernel_size', 'l1_r', 'l2_r', 'output_layer_activation', 'padding',
       'start_neurons', 'train_loss', 'train_precision', 'train_accuracy',
       'train_recall', 'val_loss', 'val_precision', 'val_accuracy',
       'val_recall', 'train_f1', 'val_f1'],
      dtype='object')